src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -177,20 +184,48 @@ protected override SynthesizedAttributeData TrySynthesizeIsReadOnlyAttribute()
             return base.TrySynthesizeIsReadOnlyAttribute();
         }
 
+        protected override SynthesizedAttributeData TrySynthesizeIsByRefLikeAttribute()
+        {
+            if ((object)_lazyIsByRefLikeAttribute != null)
+            {
+                return new SynthesizedAttributeData(
+                    _lazyIsByRefLikeAttribute.Constructor,
+                    ImmutableArray<TypedConstant>.Empty,
+                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+            }
+
+            return base.TrySynthesizeIsByRefLikeAttribute();
+        }
+
         private void CreateEmbeddedAttributesIfNeeded(DiagnosticBag diagnostics)
         {
             if (this.NeedsGeneratedIsReadOnlyAttribute)
             {
-                CreateEmbeddedAttributeIfNeeded(
-                    ref _lazyEmbeddedAttribute,
-                    diagnostics,
-                    AttributeDescription.CodeAnalysisEmbeddedAttribute);
+                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
 
                 CreateEmbeddedAttributeIfNeeded(
                     ref _lazyIsReadOnlyAttribute,
                     diagnostics,
                     AttributeDescription.IsReadOnlyAttribute);
             }
+
+            if (this.NeedsGeneratedIsByRefLikeAttribute)
+            {
+                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
+
+                CreateEmbeddedAttributeIfNeeded(
+                    ref _lazyIsByRefLikeAttribute,
+                    diagnostics,
+                    AttributeDescription.IsByRefLikeAttribute);
+            }
+        }
+
+        private void CreateEmbeddedAttributeItselfIfNeeded(DiagnosticBag diagnostics)
```

Comment:
Then I need to pass the _lazyEmbeddedAttribute in two places now. And I already done a mistake of passing a wrong _lazy . I think it is harder to get wrong when it is a separate method

---
In reply to: [118792352](https://github.com/dotnet/roslyn/pull/19795#discussion_r118792352) [](ancestors = 118792352)